eo-maven-plugin/src/main/java/org/eolang/maven/RegisterMojo.java

```diff
@@ -108,50 +108,56 @@ public final class RegisterMojo extends SafeMojo {
 
     @Override
     public void exec() throws IOException {
-        final Pattern pattern = Pattern.compile("^[a-zA-Z0-9\\-]+\\.eo$");
-        final int before = this.scopedTojos().size();
-        if (before > 0) {
-            Logger.info(this, "There are %d EO sources registered already", before);
-        }
-        final Collection<Path> sources = new Walk(this.sourcesDir.toPath())
-            .includes(this.includeSources)
-            .excludes(this.excludeSources);
-        final Unplace unplace = new Unplace(this.sourcesDir);
-        for (final Path file : sources) {
-            if (this.strictFileNames && !pattern.matcher(file.getFileName().toString()).matches()) {
-                throw new IllegalArgumentException(
-                    String.format(
-                        "Incorrect name found: '%s'. EO name must match '%s'",
-                        file.getFileName().toString(),
-                        pattern
-                    )
-                );
-            }
-            final String name = unplace.make(file);
-            if (this.scopedTojos().contains(name)) {
-                Logger.debug(this, "EO source %s already registered", name);
-                continue;
+        if (this.sourcesDir == null) {
```

Comment:
@volodya-lombrozo check this: https://www.yegor256.com/2015/01/21/if-then-throw-else.html

Reply:
@yegor256 Done. Thanks